lib/merge.js

```diff
@@ -304,17 +304,28 @@ PouchMerge.collectConflicts = function (metadata) {
   return conflicts;
 };
 
-PouchMerge.rootToLeaf = function (tree) {
+// build up a list of all the paths to the leafs in this revision tree
+PouchMerge.rootToLeaf = function (revs) {
   var paths = [];
-  PouchMerge.traverseRevTree(tree, function (isLeaf, pos, id, history, opts) {
-    history = history ? history.slice(0) : [];
```

Comment:
There is a consistent trend here: anything using `traverseRevTree` tends to be slow, because 1) it creates inner functions that can't be inlined as efficiently by V8, and 2) it tends to traverse the whole rev tree even when that's not strictly necessary (in the case of this particular function, it is).